packages/terra-form/src/Select.jsx

```diff
@@ -62,7 +62,7 @@ const Select = ({
       defaultValue={defaultValue}
       className={classNames('terra-Form-select', additionalSelectFieldProps.className)}
     >
-      {choices.map(choice => <option key={choice.toString()} value={choice}>{choice}</option>)}
+      {choices.map(choice => <option key={choice.value} value={choice.value}>{choice.display}</option>)}
```

Comment:
I'm curious too. From looking here, it looks like you need both a value and onChange for a controlled select box. https://facebook.github.io/react/docs/forms.html.